parsers/FR_O.py

```diff
@@ -157,92 +169,89 @@ def fetch_data(
 
-    return_list: List[Dict[str, Any]] = []
+    production_breakdown_list = ProductionBreakdownList(logger=logger)
     for production_object in production_objects:
-        production: Dict[str, Union[float, int]] = {}
-        storage: Dict[str, Union[float, int]] = {}
+        production = ProductionMix()
+        storage = StorageMix()
         for mode_key in production_object:
             if mode_key in PRODUCTION_MAPPING:
-                if PRODUCTION_MAPPING[mode_key] in production.keys():
-                    production[PRODUCTION_MAPPING[mode_key]] += production_object[
-                        mode_key
-                    ]
-                else:
-                    production[PRODUCTION_MAPPING[mode_key]] = production_object[
-                        mode_key
-                    ]
-                # Set values below 0 to 0
-                production[PRODUCTION_MAPPING[mode_key]] = (
-                    production[PRODUCTION_MAPPING[mode_key]]
-                    if production[PRODUCTION_MAPPING[mode_key]] >= 0
-                    else 0
+                production.add_value(
+                    PRODUCTION_MAPPING[mode_key],
+                    production_object[mode_key],
+                    correct_negative_with_zero=True,
                 )
             elif mode_key in STORAGE_MAPPING:
-                if STORAGE_MAPPING[mode_key] in storage.keys():
-                    storage[STORAGE_MAPPING[mode_key]] += (
-                        production_object[mode_key] * -1
-                    )
-                else:
-                    storage[STORAGE_MAPPING[mode_key]] = (
-                        production_object[mode_key] * -1
-                    )
-        return_list.append(
-            {
-                "zoneKey": zone_key,
-                "datetime": datetime.fromisoformat(production_object[f"{date_string}"]),
-                "production": production,
-                "storage": storage,
-                "source": "edf.fr",
-                # TODO: Should be re-enabled when the changes discussed in #4828 are implemented
-                # "estimated": True if production_object["statut"] == "Estimé" else False,
-            }
+                storage.add_value(
+                    STORAGE_MAPPING[mode_key], -production_object[mode_key]
+                )
+            elif mode_key in IGNORED_VALUES:
+                pass
+            else:
+                logger.warning(
+                    f"Unknown mode_key: '{mode_key}' encountered for {zone_key}."
+                )
+
+        production_breakdown_list.append(
+            zoneKey=zone_key,
+            datetime=datetime.fromisoformat(production_object[f"{date_string}"]),
+            production=production,
+            storage=storage,
+            source="edf.fr",
+            sourceType=EventSourceType.estimated
+            if production_object["statut"] == "Estimé"
+            else EventSourceType.measured,
```

Comment:
I don't know if this is actually supported for external estimations so if it's not I can revert it (or add a enum for estimated_external).

Reply:
It's exactly for this kind of parser that I had this in mind so it's supported :)
The only thing is that there's no action on it in the backend for now.

Reply:
Yay! Last time I added it the whole parser broke... 😅